src/bindings/jna/libelektra4j/src/main/java/org/libelektra/exception/ConflictingStateException.java

```diff
@@ -0,0 +1,16 @@
+package org.libelektra.exception;
```

Comment:
These files should be generated from the specification.

Reply:
I fear that autogeneration is not possible.
The main reason is the class hierarchy which maps to the error code.

I have also mapped invisible classes (eg. InstallationException) which do not appear at all in the specification file. These classes are neccessary though so developers can catch the more generic exception which was one of the main intentions behind the error code design.

Furthermore the autogeneration would need to know which error code is below which for correct inheritence. So there would be error prone recursive parsing of the individual digits.
Additionally we would completely deny ourself special implementation in subclasses, e.g., the ConflictingState/ OutOfMemory Exception which have more of a special meaning. All classes would be equally with autogeneration.

Reply:
> I fear that autogeneration is not possible.

Then the specification is not like it should be. We want the specification useful for several languages.

> All classes would be equally with autogeneration.

It is always possible to specialize, like I also suggested in  #2697

Reply:
The specification is useful for all languages. It is just that the specification *file* does not have all errors.
Autogeneration of C code to an exact equivalent optimized java code is very error prone. Especially when it comes to inheritance. There are actually [companies](https://www.mtsystems.com/) who specialize in this matter and still are not able to do it perfectly. Even from java to kotlin or vice versa it does not work well and they both rely on the jvm.

Implementing your desired autogeneration would be worth at least another bachelor thesis.

Reply:
> The specification is useful for all languages.

This is to be shown by you.

> It is just that the specification file does not have all errors.

Then how can it be useful?

> Especially when it comes to inheritance.

You maybe need to add the relevant information to the specification.

> Autogeneration [...]

You are talking something very offtopic here and use wrong terms. Nobody requested to write a transpiler.

The code (the exceptions) should be generated from the (one and only) specification src/error/specification

> Implementing your desired autogeneration would be worth at least another bachelor thesis.

If this would be the case then something somewhere went terribly wrong. In one bachelor thesis two complete bindings were written. The new specification should make it easier, not harder.

We also need to write some generator for Go and Rust.

Reply:
> This is to be shown by you.

It is shown here that it works for both java and C very well which can be seen in this PR. It is just that autogeneration is very complicated and error prone.

> Then how can it be useful?

Because developers are only permitted to use those error codes which are relevant and not those which are too generic. On could also write down additional codes with a metatag that says it is not allowed to use but only for "documentation" but this is really just unnecessarily bloating of the specification file.

> The code (the exceptions) should be generated from the (one and only) specification src/error/specification

Lets talk about this in another elektra meeting as it will simply be too time intensive writing here all the time.

> If this would be the case then something somewhere went terribly wrong. In one bachelor thesis two complete bindings were written. The new specification should make it easier, not harder.
We also need to write some generator for Go and Rust.

Did these bindings include autogeneration and exception mapping for a language which has the same sort of inheritance as java? Go and Rust have a different form of inheritence which is easier to model

Reply:
Lets talk about this in an elektra meeting as this would otherwise be more time intensive than needed

Reply:
Just to clarify what I am talking about: The specification in src/error/specification should be complete enough, that the relation (or subtyping, inheritance, ..) between the error codes can easily be known within a tool that generates exception classes (or needs to know if one error is subtype of another). If you find the encoding as error code letters too difficult yourself, why not encode it in an easier form?

> Did these bindings include autogeneration and exception mapping for a language which has the same sort of inheritance as java?

Yes, it used code generation far beyond of what we are talking here. Generating these few exception classes is trivial compared to that. Feel free to have a look at src/bindings/swig/python/kdb.i

> Lets talk about this in an elektra meeting as this would otherwise be more time intensive than needed

We can also talk about it but we also should document how we came to the conclusion.

And it is an "urgent" matter because:

1. you want to finish
2. the decision is needed for Go and Rust bindings which are finished soon, too

I started  #2871.

Reply:
Since this is discussed in #2871 it can stay now as it is